src/sage/schemes/elliptic_curves/homset.py

```diff
@@ -0,0 +1,151 @@
+
+EXAMPLES:
+
+The only useful thing this class does at the moment is coercing integers into
+the endomorphism ring as scalar multiplications::
+
+    sage: E = EllipticCurve([1,2,3,4,5])
+    sage: f = End(E)(7); f
+    Scalar-multiplication endomorphism [7] of Elliptic Curve defined by y^2 + x*y + 3*y = x^3 + 2*x^2 + 4*x + 5 over Rational Field
+    sage: f == E.scalar_multiplication(7)
+    True
+
+::
+
+    sage: E = EllipticCurve(GF(431^2), [0,1])
+    sage: E.automorphisms()[0] == 1
+    True
+    sage: E.automorphisms()[1] == -1
+    True
+    sage: omega = E.automorphisms()[2]
+    sage: omega == 1
+    False
+    sage: omega^3 == 1
+    True
+    sage: (1 + omega + omega^2) == 0
+    True
+    sage: (2*omega + 1)^2 == -3
+    True
+
+AUTHORS:
+
+- Lorenz Panny (2023)
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2023 Lorenz Panny
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.rings.integer_ring import ZZ
+from sage.categories.morphism import Morphism
+from sage.schemes.generic.homset import SchemeHomset_generic
+
+
+class EllipticCurveHomset(SchemeHomset_generic):
+    r"""
+    This class represents the set of all homomorphisms between two fixed
+    elliptic curves.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(419^2), [1,0])
+        sage: E.frobenius_isogeny() in End(E)
+        True
+        sage: phi = E.isogenies_prime_degree(7)[0]
+        sage: phi in End(E)
+        False
+        sage: phi in Hom(E, phi.codomain())
+        True
+
+    Note that domain and codomain are *not* taken up to isomorphism::
+
+        sage: iso = E.isomorphism_to(EllipticCurve(GF(419^2), [2,0]))
+        sage: iso in End(E)
+        False
+    """
+    def __init__(self, *args, **kwds):
+        r"""
+        Construct the homset for a given pair of curves.
+
+        TESTS::
+
+            sage: E = EllipticCurve('37a1')                                             # needs sage.schemes
+            sage: Hom(E, E).__class__                                                   # needs sage.schemes
+            <class 'sage.schemes.elliptic_curves.homset.EllipticCurveHomset_with_category'>
+
+        ::
+
+            sage: E1 = EllipticCurve(j=42)
+            sage: E2 = EllipticCurve(j=43)
+            sage: Hom(E1, E2)
+            Group of elliptic-curve morphisms
+              From: Elliptic Curve defined by y^2 = x^3 + 5901*x + 1105454 over Rational Field
+              To:   Elliptic Curve defined by y^2 + x*y = x^3 + x^2 + 1510*x - 140675 over Rational Field
+        """
+        super().__init__(*args, **kwds)
```

Comment:
It would be good to manipulate the category to know that the homset is also in `Groups()` if that is indeed the case. Although I have trouble believe that, in which case it should be the "Set of...".